src/reducers/index.js

```diff
@@ -26,30 +12,30 @@ const combinedReducer = combineReducers({
  *
  * @return {object} new state
  */
-const calculatePortsPosition = (state, action, oldState) => {
-    let node;
-    // TODO: NOT a big fan of this way to optimize port recalculations, don't feel future proof
-    if ((/FLOWDESIGNER_NODE_/.exec(action.type) && action.type !== 'FLOWDESIGNER_NODE_REMOVE') ||
-    (/FLOWDESIGNER_PORT_/.exec(action.type) && action.type !== 'FLOWDESIGNER_PORT_REMOVE')) {
-        if (action.nodeId) {
-            node = state.nodes.get(action.nodeId);
-        } else if (action.portId) {
-            node = state.nodes.get(state.ports.get(action.portId).nodeId);
-        } else {
-            invariant(false, `can't process calculatePortsPosition on ${action.type}`);
-        }
-    }
-    let newPortsPosition = new Map();
-    if (node) {
-        const ports = state.ports.filter(port => port.nodeId === node.id);
-        const calculatePortPosition = state.nodeTypes
-            .getIn([node.nodeType, 'component'])
-            .calculatePortPosition;
-        newPortsPosition = newPortsPosition
-            .merge(calculatePortPosition(ports, node.position, node.nodeSize));
-    }
-    const newState = { ...state, ports: state.ports.merge(newPortsPosition) };
-    return newState;
+const calculatePortsPosition = (state, action) => {
```

Comment:
I don't like finding code in folder/index.js
only import/export stuff
you should move that code to a position.js or 'common.js' @talend-glorieux ?